### DISCUSSION TITLE
More verbose Treesitter source output for JSON files

### DISCUSSION THREAD

**Discussion by jborbik** [Bekaboo/dropbar.nvim#36] [Q&A] [answered]
Hello,
Thank you so much for an amazing plugin which works great out of the box.
I wanted to ask if there is a way to make the JSON breadcrumbs work nicely with treesitter source. With the default treesitter source behavior the winbar only shows the name of the node the cursor is currently on. I would love to see whole structure up to the cursor position with treesitter, similarly like dropbar does it with LSP:
![Screenshot from 2023-06-14 13-41-22](https://github.com/Bekaboo/dropbar.nvim/assets/14967831/426ebfe7-bc89-4d3e-bf9f-ed3f077cf2bd)
Is this possible? Please let me know if I should provide more details.

**Comment by Bekaboo** [comment, marked as answer]
@Jendker Sorry I'm replying late, I tested with several json files and seems that the cause of the problem is the treesitter parse does not return the "parent" key of the current node as a parent treesitter node. So the only way to improve the result is to improve the treesitter parsers so that dropbar can get the parent of current json key from the parsers. There's little thing dropbar can do here.

---

EDIT: Never mind, should be fixed in commit 5c8bd1a4afe55c211c56b27a0a068ca1ff709c6e

**Comment by jborbik** [comment]
I see, thank you so much! I'll reach out to treesitter json.

EDIT: Oh, thank you so much for the fix! It works wonderfully now :) ❤️
